### ISSUE TITLE
Decharger always considers [M+H] adduct regardless of the settings

### ISSUE THREAD

**Issue by iprotsyuk** [OpenMS/OpenMS#2083]
Decharger produces wrong results with LC-MS data acquired in negative mode as it's always assumes high probability for protonated adducts even when it's not in the list of possible adducts.

**Comment by timosachsenberg** [comment]
@cbielow any take on this problem?

**Comment by cbielow** [comment]
the tool was written for proteomics, and thus has internal checks which enforce positive charges.
So without (extensive) changes to the code, I'm afraid this won't work.

**Comment by poshul** [comment]
I've been working on this problem, and have a solution that I built off of an earlier version of the OpenMS code-base. I will work on merging the latest from develop and submitting it as a pull request.

**Comment by timosachsenberg** [comment]
sound great. Think @FabianAicheler would also be interested in taking a look at it

**Comment by hendrikweisser** [comment]
@poshul:
> I will work on merging the latest from develop and submitting it as a pull request.

Has this happened? Can the issue be closed?

**Comment by poshul** [comment]
Yes, I believe so. @iprotsyuk have you had any issues with this in the last year or so?

**Comment by iprotsyuk** [comment]
Hi @poshul, I haven't used this functionality for over a year now so can't confirm really. I guess you can close the ticket for the time being

**Comment by FabianAicheler** [comment]
The MetaboliteAdductDecharger (derived from the Decharger) has support for negative mode. There, negative mode 'default adduct' is proton loss.

Not sure if we simply want to phase out the original Decharger into the Util section/remove it completely, currently no time to streamline/integrate newer stuff from MAD into the old tool.
